webapp/src/app/todo/todo.component.ts

```diff
@@ -1,14 +1,78 @@
 import { Component } from '@angular/core';
+import { Todo } from '../shared/sdk/models/Todo';
+import { TodoService } from './todo.service';
+import { FormService } from '../ui/form/ui-form.service';
 
 @Component({
   selector: 'app-todo',
-  template: `
-    <app-card title="Todos">
-      <app-todo-list></app-todo-list>
-    </app-card>
-  `,
+  templateUrl: './todo.component.html',
 })
 
 export class TodoComponent {
 
+  private todo: Todo = new Todo();
+  private createConfig;
+  private updateConfig;
+
+  private createButton = {
+    class: 'btn btn-primary float-right',
+    icon: 'plus',
+    text: 'Create'
+  };
+
+  private tableHeaders = [
+    'Text',
+    'Due',
+    'Done?',
+    'Actions',
+  ];
+
+  constructor(
+    private todoService: TodoService,
+    private formService: FormService
+  ) {
+    this.createConfig = this.createFormConfig();
+    this.updateConfig = this.updateFormConfig();
+  }
+
+  public createFormConfig() {
+    return {
+      fields: this.createFormFields,
+      showCancel: true,
+      action: 'create'
+    }
+  }
+
+  public createFormFields = [
```

Comment:
`createFormConfig` + `createFormFields` and `updateFormConfig` + `updateFormFields` are very similar, they should be combined.


Something like: 
```typescript
public formConfig(type) {
  return {
    fields: this.formFields(type),
    showCancel: true,
    action: (type === 'edit') ? 'update' : 'create'
  }
}

formFields(type) {
  const fields = [
    this.formService.input('text', {
      label: 'Text',
      placeholder: 'Text'
    }),
    this.formService.date('dueAt', {
      label: 'Due',
    }),
  ]

  if (type === 'edit') {
    fields.push(
      this.formService.checkbox('done', { label: 'Done?' }),
    )
  }
}
```